src/WebAppDIRAC/WebApp/handler/ProxyManagerHandler.py

```diff
@@ -108,93 +107,65 @@ def web_deleteProxies(self):
+      return str(week) + " weeks"
+
     day, hours = divmod(sec, 86400)
-    if day > 0:
-      if day == 1:
-        return "One day"
-      else:
-        return str(day) + " days"
+    if day == 1:
+      return "One day"
+    elif day > 0:
+      return str(day) + " days"
 
   def __request(self):
     gLogger.info("!!!  PARAMS: ", str(self.request.arguments))
+    start = int(self.get_argument("start", "0"))
+    limit = int(self.get_argument("limit", "25"))
     req = {}
 
-    start = 0
-    limit = 25
-
-    if "limit" in self.request.arguments and len(self.request.arguments["limit"][0]) > 0:
-      limit = int(self.request.arguments["limit"][0])
-
-    if "start" in self.request.arguments and len(self.request.arguments["start"][0]) > 0:
-      start = int(self.request.arguments["start"][0])
-
-    try:
-      sortDirection = str(self.request.arguments['sortDirection']).strip()
-    except BaseException:
-      sortDirection = "ASC"
-    try:
-      sortField = str(self.request.arguments['sortField']).strip()
-    except BaseException:
-      sortField = "UserName"
+    sortDirection = self.get_argument("sortDirection", "ASC")
+    sortField = self.get_argument("sortField", "UserName")
     sort = [[sortField, sortDirection]]
     gLogger.info("!!!  S O R T : ", sort)
 
-    if "username" in self.request.arguments:
-      users = list(json.loads(self.request.arguments['username'][-1]))
-      if len(users) > 0:
-        req['UserName'] = users
-
-    if "usergroup" in self.request.arguments:
-      usersgroup = list(json.loads(self.request.arguments['usergroup'][-1]))
-      if len(usersgroup) > 0:
-        req['UserGroup'] = usersgroup
-
-    if "usersgroup" in self.request.arguments and len(self.request.arguments["persistent"]) > 0:
-      if str(self.request.arguments["persistent"]) in ["True", "False"]:
-        req["PersistentFlag"] = str(self.request.arguments["persistent"])
-    before = False
-    after = False
-    if "expiredBefore" in self.request.arguments and len(self.request.arguments["expiredBefore"]) > 0:
-      try:
-        before = int(self.request.arguments["expiredBefore"])
-      except BaseException:
-        pass
-    if "expiredAfter" in self.request.arguments and len(self.request.arguments["expiredAfter"]) > 0:
-      try:
-        after = int(self.request.arguments["expiredAfter"])
-      except BaseException:
-        pass
-    if before and after:
-      if before > after:
-        req["beforeDate"] = before
-        req["afterDate"] = after
-    else:
-      if before:
-        req["beforeDate"] = before
-      if after:
-        req["afterDate"] = after
+    users = list(json.loads(self.get_argument("username", "[]")))
+    if users:
+      req['UserName'] = users
+
+    usersgroup = list(json.loads(self.get_argument("usergroup", "[]")))
+    if usersgroup:
+      req['UserGroup'] = usersgroup
+
+    persistent = self.get_argument("persistent", "")
+    if persistent in ["True", "False"]:
```

Comment:
Is there a reason to change the logic here?
```suggestion
    if usersgroup and persistent in ["True", "False"]:
```

Reply:
Well spotted! I hadn't noticed the structure was inconsistent with other similar lines.

I'm not sure if this was a bug as the original code used both `"usergroup"` and `"usersgroup"` and `"usersgroup"` doesn't appear anywhere else in the code:

```python
    if "usergroup" in self.request.arguments:
      usersgroup = list(json.loads(self.request.arguments['usergroup'][-1]))
      if len(usersgroup) > 0:
        req['UserGroup'] = usersgroup

    if "usersgroup" in self.request.arguments and len(self.request.arguments["persistent"]) > 0:
      if str(self.request.arguments["persistent"]) in ["True", "False"]:
        req["PersistentFlag"] = str(self.request.arguments["persistent"])
```

and most lines like this looked like:

```
    if "start" in self.request.arguments and len(self.request.arguments["start"][0]) > 0:
      start = int(self.request.arguments["start"][0])
```